diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -437,12 +394,71 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
+     */
+    @Suppress("TYPE_ALIAS", "UnsafeCallOnNullableType")
+    private fun searchSomeSplitInBinaryExpression(parent: ASTNode, configuration: LineLengthConfiguration): List<Pair<ASTNode, Int>?> {
+        val logicListOperationReference = listOf(OROR, ANDAND)
+        val compressionListOperationReference = listOf(GT, LT, EQEQ, GTEQ, LTEQ, EXCLEQ)
+        val binList: MutableList<ASTNode> = mutableListOf()
+        searchBinaryExpression(parent, binList)
+        val rightBinList = binList.map {
+            it to positionByOffset(it.getFirstChildWithType(OPERATION_REFERENCE)!!.startOffset).second
+        }
+            .sortedBy { it.second }
+            .reversed()
+        val returnList: MutableList<Pair<ASTNode, Int>?> = mutableListOf()
+        addInSmartListBinExpression(returnList, rightBinList, logicListOperationReference)
+        addInSmartListBinExpression(returnList, rightBinList, compressionListOperationReference)
+        val expression = rightBinList.firstOrNull { (it, offset) ->
+            val binExpression = it.getFirstChildWithType(OPERATION_REFERENCE)
+            offset + (it.getFirstChildWithType(OPERATION_REFERENCE)?.text!!.length ?: 0) < configuration.lineLength + 1 &&
+                    binExpression!!.firstChildNode.elementType !in logicListOperationReference &&
+                    binExpression.firstChildNode.elementType !in compressionListOperationReference &&
+                    binExpression.firstChildNode.elementType != EXCL
+        }
+        returnList.add(expression)
+        return returnList
+    }
+    @Suppress("TYPE_ALIAS", "UnsafeCallOnNullableType")
+    private fun addInSmartListBinExpression(
+        returnList: MutableList<Pair<ASTNode, Int>?>,
+        rightBinList: List<Pair<ASTNode, Int>>,
+        listTypes: List<IElementType>
+    ) {
+        val expression = rightBinList.firstOrNull { (it, offset) ->
+            val binExpression = it.getFirstChildWithType(OPERATION_REFERENCE)
+            offset + (it.getFirstChildWithType(OPERATION_REFERENCE)?.text!!.length ?: 0) < configuration.lineLength + 1 &&
+                    binExpression!!.firstChildNode.elementType in listTypes
+        }
+        returnList.add(expression)
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun searchRightSplitInBinaryExpression(parent: ASTNode, configuration: LineLengthConfiguration): Pair<ASTNode, Int>? {
+        val binList: MutableList<ASTNode> = mutableListOf()
+        searchBinaryExpression(parent, binList)
+        return binList.map {
+            it to positionByOffset(it.getFirstChildWithType(OPERATION_REFERENCE)!!.startOffset).second
+        }
+            .sortedBy { it.second }
+            .reversed()
+            .firstOrNull { (it, offset) ->
+                offset + (it.getFirstChildWithType(OPERATION_REFERENCE)?.text!!.length ?: 0) <= configuration.lineLength + 1
+            }
+    }
+
     /**
```

Comment:
Is this comment connected to something?

Reply:
Removed unnecessary comments, added the necessary comments